### PR TITLE
{lib}[intel/2016b] Tensorflow v0.12.1 w/ Python 3.5.2

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2304.phanpy.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/572f5585749fce888f56e661d2e293c2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2039.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/34bb8acaeadfe4d70aeae17521188dc2 for a full test report.

File: easybuild/easyconfigs/p/protobuf/protobuf-3.2.0-intel-2016b.eb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+homepage = 'https://github.com/google/protobuf/'
+description = """Google Protocol Buffers"""
+
+toolchain = {'name': 'intel', 'version': '2016b'}
+
+source_urls = ['https://github.com/google/protobuf/archive/v%(version)s/']
+sources = [SOURCE_TAR_GZ]
+
+preconfigopts = "./autogen.sh &&"
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/protobuf/protobuf-3.2.0-intel-2016b.eb]: you need `autotools` as build dep for this.

File: easybuild/easyconfigs/t/Tensorflow/Tensorflow-0.12.1-intel-2016b-Python-3.5.2.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://www.tensorflow.org/'
+description = "An open-source software library for Machine Intelligence"
+
+toolchain = {'name': 'intel', 'version': '2016b'}
+
+pymaj = '3'
+pymin = '5'
+pymajmin = pymaj + pymin
```
Comment (wpoely86) [inline easybuild/easyconfigs/t/Tensorflow/Tensorflow-0.12.1-intel-2016b-Python-3.5.2.eb]: don't we have constants for this?

**Comment by rjeschmi** [comment]
I can't test it easily right now, but it looks ok in theory.

Do you think it might be worthwhile to offer a compiled version?

**Comment by boegel** [comment]
@rjeschmi I usually prefer building from source, but I just don't have the time to dive into that now for Tensorflow. It certainly couldn't hurt, imho.

The build instructions are at https://www.tensorflow.org/get_started/os_setup#installing_from_sources; they require a custom build tool `bazel` (we have an easyconfig using the `CrayGNU` toolchain for that), and you obtain a custom `.whl` that you can use to install `Tensorflow`.

So, that's most likely a custom easyblock...

I think @gppezzi has some experience with building `Tensorflow` from source.
Is that effort publicly available somewhere @gppezzi?

**Comment by boegel** [comment]
Hah, the `configure` script is an interactive script, so *definitely* a custom easyblock...
Cfr. https://www.tensorflow.org/get_started/os_setup#configure_the_installation

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2304.phanpy.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/fd6be5712d54650066d08b5e6f1fabec for a full test report.

**Comment by gppezzi** [comment]
We did already build TF from sources, we have an open [PR](https://github.com/eth-cscs/production/pull/40) for TF with and without Cuda on Cray (but this is WIP).

**Comment by boegel** [comment]
@gppezzi Do you have a particular reason for not using a `Tensorflow` easyblock? Seems like that would be appropriate here...

**Comment by gppezzi** [comment]
I just don't think that it is worth the efforts right now: TF seems to be changing very quickly and, by time we manage to have a working easyblock, things will change. 

IMO chances of re-using the easyblock for a new version (or on a different system) are very low. Also, we need more time to better understand this build tool (the procedure is currently a huge mess).

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2039.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/6e17d6be65d78d390cba1d95ae3e49f9 for a full test report.

**Comment by verdurin** [comment]
@boegel why is the `foss-2016a` version of `protobuf` included here?

**Comment by boegel** [comment]
@verdurin I wanted to make the change requested by @wpoely86  (incl. `Autotools` as a build dep) consistent for `protobuf` easyconfigs that call `autogen.sh`

**Comment by boegel** [comment]
Thanks for the feedback @rjeschmi and @gppezzi.

I'll go ahead and merge this as-is, we can re-evaluate building from source later.

It seems like TF is close to a 1.0 release, which suggests that the build procedure should be stable too...

**Pull request merged**